lib/matplotlib/style/core.py

```diff
@@ -66,22 +67,92 @@ def use(style):
     else:
         styles = style
 
+    styles = insert_parent_styles(styles)
+
     for style in styles:
         if not cbook.is_string_like(style):
             mpl.rcParams.update(style)
             continue
 
-        if style in library:
-            mpl.rcParams.update(library[style])
+        mpl.rcParams.update(get_style_parameters(style))
+
+
+def insert_parent_styles(styles):
+    """Returns a list with all the styles and parents styles called"""
+    tmp_styles = []
+    for style in styles:
+        style = _get_parent_styles_from_str_or_dict(style)
+        tmp_styles.extend(style)
+    return tmp_styles[:]
+
+
+def _get_parent_styles_from_str_or_dict(styles):
+    """Returns the parent styles from a single style or a dict"""
+    styles = [styles]
+    has_parent = True
+    visited_styles = []
+    temp_styles = styles[:]
+
+    while has_parent:
+        for style in styles:
+            if style in visited_styles:
+                continue
+            visited_styles.append(style)
+
+            if _is_dict_like(style):
+                style_dic = style
+            else:
+                style_dic = get_style_parameters(style)
+
+            has_parent = 'style' in style_dic
```

Comment:
I don't think this is sufficient to traverse a tree. If earlier nodes had a parent, but the last one doesn't, then those parents aren't ever visited, right?